### ISSUE TITLE
Xournal++ on Hyprland works properly only if scale is 1

### ISSUE THREAD

**Issue by awerty-noob** [hyprwm/Hyprland#6234]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch main at commit ce17961aad6f9164e5d026d19efd42b07b123bff dirty (keybinds: Added new dispatcher (sendshortcut) (6174)).
Date: Sat May 25 02:58:26 2024
Tag: v0.40.0-131-gce17961a, commits: 4737

flags: (if any)


System Information:
System name: Linux
Node name: archlinux
Release: 6.9.1-arch1-2
Version: #1 SMP PREEMPT_DYNAMIC Wed, 22 May 2024 13:47:07 +0000


GPU information: 
64:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Phoenix1 [1002:15bf] (rev c2) (prog-if 00 [VGA controller])


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:

```
</details>

### Bug or Regression?

Bug

### Description

The option "View -> Highlight cursor position" often, but not always, produces a pencil icon on a position different from where Xournal++ is actually writing.
![problem](https://github.com/hyprwm/Hyprland/assets/36948728/71a80afe-fbc2-4392-8de7-34179e18ce3e)

But this bug occurs only if the scale is not equal to 1. In other words, the cursor in right on the path only if the scale is 1. I make an issue [https://github.com/xournalpp/xournalpp/issues/5704](url) there, and this bug does not occur on Plasma Wayland with any scale, so I guess this is a bug from Hyprland. 
Btw, I ran `gtk3-demo --run=drawingarea` on Hyprland with scale 1.25, and the position of cursor is correct.

### How to reproduce

1. Install Xournal++ with latest version
2. set the scale to a non-one value (From my case, is 1.25)
3. draw something.

**Comment by sungyoonc** [comment]
Can you check git after commit 90f262aada21af9be1c5ccf89b8dc5e0d28478f1 ?

**Comment by awerty-noob** [comment]
> Can you check git after commit [90f262a](https://github.com/hyprwm/Hyprland/commit/90f262aada21af9be1c5ccf89b8dc5e0d28478f1) ?

I compiled from commit 553232a, the bug still exists.

**Comment by awerty-noob** [comment]
Fixed in df6ebe358b30ee7b49f296e05763e5e4b0edce98, Closing.
